metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -497,10 +541,63 @@ public Map<ConfigResource, ResultOrError<Map<String, String>>> describeConfigs(
         return results;
     }
 
+    // The function is used when enabling ELR. It will create a new cluster level min ISR config if there is not any.
+    // Also, it will remove all the broker level min ISR config records.
+    void maybeResetMinIsrConfig(List<ApiMessageAndVersion> outputRecords) {
+        if (!clusterConfig().containsKey(MIN_IN_SYNC_REPLICAS_CONFIG)) {
+            String minIsrDefaultConfigValue = configSchema.getStaticOrDefaultConfig(
```

Comment:
We should log a message if we are doing this. Also, it seems easier just to create the configuration record directly than call a function here.

Reply:
Sounds good, updated.